### PR TITLE
Updates where work related preservation_events occur

### PR DESCRIPTION
**Pull request by devanshu-m**
* This commit overrides the `create` method from Hyrax::Actors::BaseActor
and adds three preservation_events that occur immediately after
work creation.
* Also, removes create_work_job and its spec. We don't need the
modifications in those files anymore since the create_work_job is
used in the batch_create_job.
* Edits the preservation event create method so that now we save without
the bang in the end. This will prevent from trying to run validations
again.

### CODE REVIEW

File: spec/actors/hyrax/actors/curate_generic_work_actor_spec.rb [hunk 1]
```diff
@@ -1,9 +1,55 @@
...
+    let(:uploaded_file) { Hyrax::UploadedFile.create(file: file, user: user) }
+    let(:terminator) { Hyrax::Actors::Terminator.new }
+
+    before do
+      allow(terminator).to receive(:create).and_return(true)
+    end
+
+    context 'success' do
+      before do
+        redlock_client_stub
```
Comment (rotated8) [inline spec/actors/hyrax/actors/curate_generic_work_actor_spec.rb]: Let's try the test without this. Circle should have Redis, so I don't think we need to stub the connection.

File: app/lib/preservation_events.rb [hunk 2]
```diff
@@ -13,6 +13,6 @@ def create_preservation_event(object, event)
                                               initiating_user:  event['user'],
                                               outcome:          event['outcome'],
                                               software_version: event['software_version'] }]
-    object.save!
+    object.save
```
Comment (rotated8) [inline app/lib/preservation_events.rb]: Please open a ticket pointing to this diff so we can fix the affected tests.

File: spec/actors/hyrax/actors/curate_generic_work_actor_spec.rb (hunk 1 above)
Comment (devanshu-m) [reply spec/actors/hyrax/actors/curate_generic_work_actor_spec.rb]: Removed redlock client stub

File: app/lib/preservation_events.rb (hunk 2 above)
Comment (devanshu-m) [reply app/lib/preservation_events.rb]: Replaced failing system test with unit test to restore the bang save.

**Pull request merged**